Add potency information to chat card

cdverrett94 opened the pull request:
Closes #177

Summary of changes:
- Updated potency property to a schema that includes `enabled`, `characteristic`, and `value`. Value has an initial value of the relevant potency values roll data (i.e. @potency.weak).
- Added a `getPotencyData` method for retrieving the potency data for a given tier. This will evaluate the potency value and generate the embed text.
- Added the ability uuid to the roll data, so the ability info can be pulled during the roll's context preparation.
- The roll chat card now displays the potency embed text and whether the given target passes the potency value.
- I also updated the chat card to remove the duplicate formulas. Now the base power roll is displayed and for each target/roll only the total is shown. This will save space if there are many targets.
- Random bug fix for ability distance's not localizing the tertiary values.

Updated chat card
![update-chat-card](https://github.com/user-attachments/assets/3ded04e6-79c3-474b-bcce-a29e1323be7f)

JPMeehan commented:
What's the value of checking enabled vs. seeing if characteristic has something set?

cdverrett94 commented:
I think for enabled, cleaner display on the sheet to hide the fields and a more clear check than potency.characteristic !== "". But I suppose I'm not totally attached to the idea.

I'm okay with making the potency characteristic a top level property. In that case, I think enabled makes more sense as well because there are examples where only 1 tier has an effect that has a potency.

JPMeehan commented:
Good spot on only some tiers having potency, keep enabled and just have the top level characteristic choice. I think the potency stuff should be hidden on the tiers if there's no characteristic set

cdverrett94 commented:
While I'm in this, I'm wondering. Should the potency stuff be a part of the AE stuff? Like have the `ae` be a `SchemaField` with `effects` and `potency`? Is there a situation where there would be a potency but not an AE as well?

JPMeehan commented:
So the tricky part is you can have
- AE without potency 
- AE strictly from potency
- Split AE that's always applied + more with potency

cdverrett94 commented:
Good point. Took me a while of searching through the hero doc to find those examples, but I see them. I'll leave it as is.

I updated to have a top level `potencyCharacteristic`. In `getPotencyData` I updated it to only send the full data if the potency is enabled and there is a characteristic set.

JPMeehan commented:
Monster doc is also important and frequently does weirder things

JPMeehan commented on src/module/data/item/ability.mjs:
```diff
@@ -174,6 +178,31 @@ export default class AbilityModel extends BaseItemModel {
+
+    if (!potencyData.enabled) return potencyData;
+
+    const potencyValue = (await new DSRoll(potency.value, this.parent.getRollData()).evaluate()).total;
```
Can't we just do evaluateSync here, since potency.value is deterministic

JPMeehan reviewed (approved):
Some small details to fix - I'm also wondering if we can improve the styles on the potency display in the chat card, definitely want to add some <span> pieces to wrap things for targeting styles later.

cdverrett94 commented:
> Some small details to fix - I'm also wondering if we can improve the styles on the potency display in the chat card, definitely want to add some pieces to wrap things for targeting styles later.

I've wrapped the individual components into their own spans so they can be targeted with CSS. Any specific ideas on styling it better?

JPMeehan commented:
> I've wrapped the individual components into their own spans so they can be targeted with CSS. Any specific ideas on styling it better?

Bolding `Potency:`, and I feel like some kind of distinction for success vs. failure is justified here?

cdverrett94 commented:
![potency-styling](https://github.com/user-attachments/assets/4005ffd0-2f6e-47d7-b314-048bd12f5f1a)

Styling per suggestions.

JPMeehan commented:
Says there are merge conflicts

cdverrett94 commented:
Resolved that!
